### PR TITLE
Set tpm_enabled flag according to OS requirements

### PR DESCRIPTION
**Pull request by rszwajko**
Fixes: #1596
Depends on: https://github.com/oVirt/ovirt-engine/pull/514

For both new and edited VMs:
1. set `tpm_enabled = true` if TPM support is required by the guest OS

For edited VMs:
1. set `tpm_enabled = false` if TPM is not supported by the guest OS.
2. require confirmation before disabling TPM as it may result in data
   loss.

### CODE REVIEW

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by rszwajko** [comment]
### TPM flag on Summary step (Create VM Wizard)

image

### Confirmation message warning about potential data loss

image

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Review by sjd78 — APPROVED** [review]
- transform and transport changes LGTM
 - create wizard changes LGTM
 - VM details card OS change driven TPM change confirm modal changes LGTM

**Review by sgratch — CHANGES_REQUESTED** [review]
@rszwajko 
1. The following exception occurs for the following use case:
    - edit an existing vm and change the OS from a required TPM OS to a non TPM supported OS (e.g. change OS from 2022 to Linux).
    - the confirmation dialog tpmNotSupportedByTheSystem appears.
    - click the ok button
    - the following exception appears:
       
       image
       image

2. When changing OS from a required TPM OS to a non TPM supported OS (e.g. change OS from 2022 to Linux) and then clicking on `cancel` button once the confirmation dialog `tpmNotSupportedByTheSystem` appears, from now on changing to any TPM supported OS (e.g. Window 2008) still causes the confirmation dialog `tpmNotSupportedByTheSystem` to appear. Only after exiting the vm details card edit mode, the confirmation dialog stopped appearing

3. Need to consider supporting cases in which vm.cluster level >= 4.6 and vm.TPM-required OS (win 11/2022) and vm.Chipset type <> UEFI. In those cases the Chipset type should be changed automatically without notifying the user to UEFI. same as supported on webadmin. This can be done on a follow up fix.

**Comment by rszwajko** [comment]
@sgratch 
fixed the scenarios pointed above - please review again.

File: src/components/CreateVmWizard/helpers.js [hunk 1]
```diff
@@ -163,12 +165,32 @@ const verifyOsIdToCluster = (selectedOsId, clusterId, { clusters, operatingSyste
   return selectedOs ? selectedOs.get('id') : '0'
 }
 
+const isTpmRequired = (operatingSystemId, operatingSystems) => {
+  return operatingSystems.getIn([operatingSystemId, 'tpmSupport']) === 'required'
+}
+
+const getTpmChange = (operatingSystemId, operatingSystems) => {
```
Comment (sgratch) [inline src/components/CreateVmWizard/helpers.js:172]: Why declaring this method here and not in src/helpers.js.
It is not really used by the createVmWizard.

**Review by sgratch — COMMENTED** [review]
@rszwajko 
Issues 1, 2 are fixed and looks good now.
For issue 3 - a followup fix will be created.

File: src/components/CreateVmWizard/helpers.js (hunk 1 above)
Comment (rszwajko) [reply src/components/CreateVmWizard/helpers.js:172]: no particular reason - we can change it

**Review by sgratch — APPROVED** [review]

**Comment by sgratch** [comment]
OST failed 3 times in test_verify_add_hosts (didn't get to ui tests yet) so merging without waiting for OST to succeed.

**Pull request merged**